## JSON 만들기

In [154]:
#-*-coding: utf-8
# 시장/ 부시장 직속 부서들만 먼저!
json_hierarchy = [
    {
        "div" : "서울특별시장"
    },
    {
        "div" : "대변인",
        "child" : "언론담당관"
    },
    {
        "div": "감사위원회",
        "child": "조사담당관/안전담당관/감사담당관"
    },
    {
        "div": "시민감사옴부즈만위원회"
    },
    {
        "div": "서울혁신기획관",
        "child": "인권담당관/지역공동체담당관/사회혁신담당관/청년정책담당관/갈등조정담당관/민관협력담당관"
    },
    {
        "div": "시민소통기획관",
        "child": "시민소통담당관/시민봉사담당관/도시브랜드담당관/신속행정담당관/뉴미디어담당관"
    },
    {
        "div": "행정1부시장"
    },
    {
        "div": "여성가족정책실",
        "child": "가족담당관/외국인다문화담당관/여성정책담당관/보육담당관"
    },
    {
        "div": "민생사법경찰단"
    },
    {
        "div": "비상기획관",
        "child": "민방위담당관"
    },
    {
        "div": "정보기획관",
        "child": "통계데이터담당관/정보기획담당관/정보통신보안담당관/공간정보담당관/정보시스템담당관"
    },
    {
        "div": "일자리노동정책관",
        "child": "사회적경제담당관/노동정책담당관/일자리정책담당관"
    },
    {
        "div": "행정2부시장"
    },
    {
        "div": "도시공간개선단"
    },
    {
        "div": "기술심사담당관"
    },
    {
        "div": "정무부시장"
    },
   {
       "div": "기획조정실",
       "child": [
           {
               "div": "국제교류담당관"
           },
           {
               "div": "정책기획관",
               "child": "기획담당관/법률지원담당관/대외협력담당관/조직담당관/법무담당관/평가담당관"
           },
           {
               "div": "해외도시협력담당관"
           },
           {
               "div": "재정기획관",
               "child": "예산담당관/시민참여예산과/공기업담당관/재정관리담당관/시민참여예산담당관"
           }
       ]
    },
    {
       "div": "경제진흥본부",
       "child": {
           "div": "창조경제기획관",
           "child": "도시농업과/경제정책과/민생경제과/화융합경제과/디지털창업과/투자유치과/소상공인지원과"
       }
    },
    {
          "div": "복지본부",
          "child":"자활지원과/어르신복지과/희망복지지원과/복지정책과/장애인복지정책과/인생이모작지원과/장애인자립지원과"
       },

       {
          "div": "문화본부",
          "child":[
            {
              "div": "문화정책과"
            },
            {
              "div": "한양도성도감"
            },
            {
              "div": "문화시설추진단",
              "child": "문화시설과/박물관"
            },
            {
              "div": "역사문화재과"
            },
            {
              "div": "디자인정책과"
            },
            {
              "div": "문화예술과"
            }
          ]
       },
       {
          "div": "기후환경본부",
          "child":"자원순환과/대기정책과/기후대기과/에너지시민협력과/녹색에너지과/생활환경과/대기관리과/환경정책과/기후변화대응과"
       },
       {
          "div": "행정국",
          "child":"인사과/인력개발과/정보공개정책과/총무과/자치행정과"
       },
       {
          "div": "재무국",
          "child": "38세금징수과/계약심사과/재무과/세무과/자산관리과/세제과"
       },
       {
         "div": "평생교육국",
         "child": "평생교육과/청소년정책과/교육정책과/친환경급식"
       },
       {
         "div": "관광체육국",
         "child": "관광정책과/체육진흥과/체육정책과/관광사업과"
       },
       {
         "div": "시민건강국",
         "child": "식품안전과/생활보건과/식품정책과/건강증진과/동물보호과/보건의료정책과"
       },
       {
         "div": "안전총괄본부",
         "child":[
           {
             "div":"서울역일대종합발전기획단"
           },
           {
              "div":"안전총괄관",
              "child": "안전총괄과/시설안전과/도로관리과/교량안전과/도로시설과/도로계획과/보도환경개선과/상황대응과"
           }
         ]
       },
    ]


## JSON을 이용해 url 만들기

In [160]:
from urllib import parse

# URL 추출
urls = [];
tail = '&dept%5B%5D='

# {} dict 
# [] list

def extract_value(data, origin_url):
    if(isinstance(data, list)):
        for single_list in data:
            extract_value(single_list, origin_url)
    elif(isinstance(data, dict)):
        #dict {} 일 경우
        if(len(data) == 1):
            if(origin_url != ''):
                origin_url = origin_url + tail
            div = data.get('div')
            url = origin_url + parse.quote(div)
            urls.append(url)
        else:
            div = data.get('div')
            child = data.get('child', 'empty')
            if origin_url != '':
                origin_url = origin_url + tail
            extract_value(child, origin_url + parse.quote(div))
    elif(isinstance(data, str)):
        divs = data.split('/')
        for single_div in divs:
            url = origin_url + tail + parse.quote(single_div)
            urls.append(url)

extract_value(json_hierarchy, "")

## 크롤링

In [161]:
# 필요한 라이브러리 가져오기
from urllib import request
!pip install html5lib
import html5lib
!pip install lxml
import pandas as pd
import numpy as np

In [174]:
import re
df = pd.DataFrame()
base_url_head = "https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D="
url_2017 = "&year=2017&month="
url_2018 = "&year=2018&month="
column_div = "1소속부서"

# urls돌면서 데이터 가져오기
for div in urls:
  div_url = (base_url_head + div)
  # unquote_div : url 첫부분 가져와서 
  unquote_div = parse.unquote(div)
  div_name = re.search('([가-힣]*)&de', unquote_div)
  if div_name is None:
    div_name = unquote_div
  else:
    div_name = div_name.group(1)
    
  for month in range(1,9): 
    url_month = div_url + url_2018 + str(month)
    try:
      df_month = pd.read_html(url_month)[1]
      # decode된 url의 첫 부분을 1소속부서 column에 넣어주기
      df_month[column_div] = div_name
      df = df.append(df_month)
      print(url_month)
    except IndexError:
      print('err')
    
  for month in range(8,13): 
    url_month = div_url + url_2017 + str(month)
    try:
      df_month = pd.read_html(url_month)[1]
      # decode된 url의 첫 부분을 1소속부서 column에 넣어주기
      df_month[column_div] = div_name
      df = df.append(df_month)
      print(url_month)
    except IndexError:
      print('err')
    

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%8A%B9%EB%B3%84%EC%8B%9C%EC%9E%A5&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%8A%B9%EB%B3%84%EC%8B%9C%EC%9E%A5&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%8A%B9%EB%B3%84%EC%8B%9C%EC%9E%A5&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%8A%B9%EB%B3%84%EC%8B%9C%EC%9E%A5&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%8A%B9%EB%B3%84%EC%8B%9C%EC%9E%A5&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%E

C:\Users\jeeyun\Anaconda3\lib\site-packages\pandas\core\frame.py:6201: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=True'.

To retain the current behavior and silence the warning, pass sort=False

  sort=sort)


https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%8C%80%EB%B3%80%EC%9D%B8&dept%5B%5D=%EC%96%B8%EB%A1%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%8C%80%EB%B3%80%EC%9D%B8&dept%5B%5D=%EC%96%B8%EB%A1%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%8C%80%EB%B3%80%EC%9D%B8&dept%5B%5D=%EC%96%B8%EB%A1%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%8C%80%EB%B3%80%EC%9D%B8&dept%5B%5D=%EC%96%B8%EB%A1%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%8C%80%EB%B3%80%EC%9D%B8&dept%5B%5D=%EC%96%B8%EB%A1%A0%EB%8B%B4%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B0%90%EC%82%AC%EC%98%B4%EB%B6%80%EC%A6%88%EB%A7%8C%EC%9C%84%EC%9B%90%ED%9A%8C&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B0%90%EC%82%AC%EC%98%B4%EB%B6%80%EC%A6%88%EB%A7%8C%EC%9C%84%EC%9B%90%ED%9A%8C&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B0%90%EC%82%AC%EC%98%B4%EB%B6%80%EC%A6%88%EB%A7%8C%EC%9C%84%EC%9B%90%ED%9A%8C&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B0%90%EC%82%AC%EC%98%B4%EB%B6%80%EC%A6%88%EB%A7%8C%EC%9C%84%EC%9B%90%ED%9A%8C&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%ED%98%81%EC%8B%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%ED%98%81%EC%8B%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%ED%98%81%EC%8B%A0%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%ED%98%81%EC

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EA%B4%80%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EA%B4%80%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EA%B4%80%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%84%9C%EC%9A%B8%ED%98%81%EC%8B%A0%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EA%B4%80%ED%98%91%EB%A0%A5%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8F%84%EC%8B%9C%EB%B8%8C%EB%9E%9C%EB%93%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8F%84%EC%8B%9C%EB%B8%8C%EB%9E%9C%EB%93%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8F%84%EC%8B%9C%EB%B8%8C%EB%9E%9C%EB%93%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8F%84%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%89%B4%EB%AF%B8%EB%94%94%EC%96%B4%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EC%86%8C%ED%86%B5%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%89%B4%EB%AF%B8%EB%94%94%EC%96%B4%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=12
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%951%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%951%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%951%EB%B6%80%EC%8B

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%97%AC%EC%84%B1%EA%B0%80%EC%A1%B1%EC%A0%95%EC%B1%85%EC%8B%A4&dept%5B%5D=%EC%97%AC%EC%84%B1%EC%A0%95%EC%B1%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%97%AC%EC%84%B1%EA%B0%80%EC%A1%B1%EC%A0%95%EC%B1%85%EC%8B%A4&dept%5B%5D=%EC%97%AC%EC%84%B1%EC%A0%95%EC%B1%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%97%AC%EC%84%B1%EA%B0%80%EC%A1%B1%EC%A0%95%EC%B1%85%EC%8B%A4&dept%5B%5D=%EC%97%AC%EC%84%B1%EC%A0%95%EC%B1%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%97%AC%EC%84%B1%EA%B0%80%EC%A1%B1%EC%A0%95%EC%B1%85%EC%8B%A4&dept%5B%5D=%EC%97%AC%EC%84%B1%EC%A0%95%EC%B1%85%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B9%84%EC%83%81%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EB%B0%A9%EC%9C%84%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B9%84%EC%83%81%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EB%B0%A9%EC%9C%84%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B9%84%EC%83%81%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EB%B0%A9%EC%9C%84%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B9%84%EC%83%81%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%AF%BC%EB%B0%A9%EC%9C%84%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%A0%95%EB%B3%B4%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%A0%95%EB%B3%B4%ED%86%B5%EC%8B%A0%EB%B3%B4%EC%95%88%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%A0%95%EB%B3%B4%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%A0%95%EB%B3%B4%ED%86%B5%EC%8B%A0%EB%B3%B4%EC%95%88%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%A0%95%EB%B3%B4%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%A0%95%EB%B3%B4%ED%86%B5%EC%8B%A0%EB%B3%B4%EC%95%88%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%A0%95%EB%B3%B4%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EC%A0%95%EB%B3%B4%ED%86%B5%EC%8B%A0%EB%B3%B4%EC

err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9D%BC%EC%9E%90%EB%A6%AC%EB%85%B8%EB%8F%99%EC%A0%95%EC%B1%85%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%EC%A0%81%EA%B2%BD%EC%A0%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9D%BC%EC%9E%90%EB%A6%AC%EB%85%B8%EB%8F%99%EC%A0%95%EC%B1%85%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%EC%A0%81%EA%B2%BD%EC%A0%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9D%BC%EC%9E%90%EB%A6%AC%EB%85%B8%EB%8F%99%EC%A0%95%EC%B1%85%EA%B4%80&dept%5B%5D=%EC%82%AC%ED%9A%8C%EC%A0%81%EA%B2%BD%EC%A0%9C%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9D%BC%EC%9E%90%EB%A6%AC%EB%85%B8%EB%8F%99%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2017&month=10
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%9E%A5&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%952%EB%B6%80%EC%8B%9C%EC%

err
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EA%B5%AD%EC%A0%9C%EA%B5%90%EB%A5%98%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EA%B5%AD%EC%A0%9C%EA%B5%90%EB%A5%98%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=9
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EA%B5%AD%EC%A0%9C%EA%B5%90%EB%A5%98%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EA%B5%AD%EC%A0%9C%EA%B5%90%EB%A5%98%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2017&month=12
https://op

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8C%80%EC%99%B8%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8C%80%EC%99%B8%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EB%8C%80%EC%99%B8%ED%98%91%EB%A0%A5%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%ED%8F%89%EA%B0%80%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%ED%8F%89%EA%B0%80%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%A0%95%EC%B1%85%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%ED%8F%89%EA%B0%80%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%9E%AC%EC%A0%95%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EA%B3%B5%EA%B8%B0%EC%97%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%9E%AC%EC%A0%95%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EA%B3%B5%EA%B8%B0%EC%97%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A1%B0%EC%A0%95%EC%8B%A4&dept%5B%5D=%EC%9E%AC%EC%A0%95%EA%B8%B0%ED%9A%8D%EA%B4%80&dept%5B%5D=%EA%B3%B5%EA%B8%B0%EC%97%85%EB%8B%B4%EB%8B%B9%EA%B4%80&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9A%8D%EC%A

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%90%ED%99%9C%EC%A7%80%EC%9B%90%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%90%ED%99%9C%EC%A7%80%EC%9B%90%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%90%ED%99%9C%EC%A7%80%EC%9B%90%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%90%ED%99%9C%EC%A7%80%EC%9B%90%EA%B3%BC&year=2018&month=6
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%A5%EC%95%A0%EC%9D%B8%EB%B3%B5%EC%A7%80%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%A5%EC%95%A0%EC%9D%B8%EB%B3%B5%EC%A7%80%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%A5%EC%95%A0%EC%9D%B8%EB%B3%B5%EC%A7%80%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%B3%B5%EC%A7%80%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%9E%A5%EC%95%A0%EC%9D%B8%EB%B3%B5%EC%A7%80%EC%A0%95%EC%B1%85%EA%B3%BC&year=2017&month=8
https://opengov.seou

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%AC%B8%ED%99%94%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%AC%B8%ED%99%94%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%AC%B8%ED%99%94%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%AC%B8%ED%99%94%EC%A0%95%EC%B1%85%EA%B3%BC&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%94%94%EC%9E%90%EC%9D%B8%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%94%94%EC%9E%90%EC%9D%B8%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%94%94%EC%9E%90%EC%9D%B8%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EB%AC%B8%ED%99%94%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%94%94%EC%9E%90%EC%9D%B8%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%8C%80%EA%B8%B0%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%8C%80%EA%B8%B0%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%8C%80%EA%B8%B0%EC%A0%95%EC%B1%85%EA%B3%BC&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%8C%80%EA%B8%B0%EC%A0%95%EC%B1%85%EA%B3%BC&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%85%B9%EC%83%89%EC%97%90%EB%84%88%EC%A7%80%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%85%B9%EC%83%89%EC%97%90%EB%84%88%EC%A7%80%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%85%B9%EC%83%89%EC%97%90%EB%84%88%EC%A7%80%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B8%B0%ED%9B%84%ED%99%98%EA%B2%BD%EB%B3%B8%EB%B6%80&dept%5B%5D=%EB%85%B9%EC%83%89%EC%97%90%EB%84%88%EC%A7%80%EA%B3%BC&year=2017&month=8
https://opengov.seou

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9D%B8%EC%82%AC%EA%B3%BC&year=2018&month=7
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9D%B8%EC%82%AC%EA%B3%BC&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9D%B8%EC%82%AC%EA%B3%BC&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9D%B8%EC%82%AC%EA%B3%BC&year=2017&month=10
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9D%B8%EC%82%AC%EA%B3%BC&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9E%90%EC%B9%98%ED%96%89%EC%A0%95%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9E%90%EC%B9%98%ED%96%89%EC%A0%95%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9E%90%EC%B9%98%ED%96%89%EC%A0%95%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9E%90%EC%B9%98%ED%96%89%EC%A0%95%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%96%89%EC%A0%95%EA%B5%AD&dept%5B%5D=%EC%9E%90%EC%B9%98%ED

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B5%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B3%BC&year=2017&month=12
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B5%AD&dept%5B%5D=%EC%84%B8%EB%AC%B4%EA%B3%BC&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B5%AD&dept%5B%5D=%EC%84%B8%EB%AC%B4%EA%B3%BC&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B5%AD&dept%5B%5D=%EC%84%B8%EB%AC%B4%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%9E%AC%EB%AC%B4%EA%B5%AD&dept%5B%5D=%EC%84%B8%EB%AC%B4%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B5%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B3%BC&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B5%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B3%BC&year=2017&month=10
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B5%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B3%BC&year=2017&month=12
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%ED%8F%89%EC%83%9D%EA%B5%90%EC%9C%A1%EA%B5%AD&dept%5B%5D=%EC%B2%AD%EC%86%8C%EB%85%84%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%E

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B4%80%EA%B4%91%EC%B2%B4%EC%9C%A1%EA%B5%AD&dept%5B%5D=%EC%B2%B4%EC%9C%A1%EC%A7%84%ED%9D%A5%EA%B3%BC&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B4%80%EA%B4%91%EC%B2%B4%EC%9C%A1%EA%B5%AD&dept%5B%5D=%EC%B2%B4%EC%9C%A1%EC%A7%84%ED%9D%A5%EA%B3%BC&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B4%80%EA%B4%91%EC%B2%B4%EC%9C%A1%EA%B5%AD&dept%5B%5D=%EC%B2%B4%EC%9C%A1%EC%A7%84%ED%9D%A5%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EA%B4%80%EA%B4%91%EC%B2%B4%EC%9C%A1%EA%B5%AD&dept%5B%5D=%EC%B2%B4%EC%9C%A1%EC%A7%84%ED%9D%A5%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EC%83%9D%ED%99%9C%EB%B3%B4%EA%B1%B4%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EC%83%9D%ED%99%9C%EB%B3%B4%EA%B1%B4%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EC%83%9D%ED%99%9C%EB%B3%B4%EA%B1%B4%EA%B3%BC&year=2017&month=8
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EC%83%9D%ED%99%9C%EB%B3%B4%EA%B1%B4%EA%B3%BC&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EB%8F%99%EB%AC%BC%EB%B3%B4%ED%98%B8%EA%B3%BC&year=2017&month=12
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EB%B3%B4%EA%B1%B4%EC%9D%98%EB%A3%8C%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=1
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EB%B3%B4%EA%B1%B4%EC%9D%98%EB%A3%8C%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=2
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%8B%9C%EB%AF%BC%EA%B1%B4%EA%B0%95%EA%B5%AD&dept%5B%5D=%EB%B3%B4%EA%B1%B4%EC%9D%98%EB%A3%8C%EC%A0%95%EC%B1%85%EA%B3%BC&year=2018&month=3
https://opengov.seoul.go.kr/expense?dept%5B%5

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EC%8B%9C%EC%84%A4%EC%95%88%EC%A0%84%EA%B3%BC&year=2017&month=9
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EC%8B%9C%EC%84%A4%EC%95%88%EC%A0%84%EA%B3%BC&year=2017&month=10
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EC%8B%9C%EC%84%A4%EC%95%88%EC%A0%84%EA%B3%BC&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%8F%84%EB%A1%9C%EC%8B%9C%EC%84%A4%EA%B3%BC&year=2018&month=4
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%8F%84%EB%A1%9C%EC%8B%9C%EC%84%A4%EA%B3%BC&year=2018&month=5
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%8F%84%EB%A1%9C%EC%8B%9C%EC%84%A4%EA%B3%BC&year=2018&month=6
err
err
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0

https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%B3%B4%EB%8F%84%ED%99%98%EA%B2%BD%EA%B0%9C%EC%84%A0%EA%B3%BC&year=2017&month=10
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%B3%B4%EB%8F%84%ED%99%98%EA%B2%BD%EA%B0%9C%EC%84%A0%EA%B3%BC&year=2017&month=11
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8B%9C%EB%B3%B8%EC%B2%AD&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EB%B3%B8%EB%B6%80&dept%5B%5D=%EC%95%88%EC%A0%84%EC%B4%9D%EA%B4%84%EA%B4%80&dept%5B%5D=%EB%B3%B4%EB%8F%84%ED%99%98%EA%B2%BD%EA%B0%9C%EC%84%A0%EA%B3%BC&year=2017&month=12
https://opengov.seoul.go.kr/expense?dept%5B%5D=%EC%84%9C%EC%9A%B8%EC%8

크롤링 하는데 엄청 오래걸리네...

## 데이터 확인하기

In [175]:
df.describe()

집행금액
count  2.603000e+04
mean   1.339527e+05
std    3.604628e+05
min    1.080000e+03
25%    5.000000e+04
50%    8.500000e+04
75%    1.520000e+05
max    3.623653e+07

In [176]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26041 entries, 0 to 13
Data columns (total 10 columns):
1소속부서    26041 non-null object
결제방법     25993 non-null object
구분       493 non-null object
대상인원     26030 non-null object
부서명      26029 non-null object
연번       26041 non-null object
집행금액     26030 non-null float64
집행목적     26028 non-null object
집행일시     26030 non-null object
집행장소     26030 non-null object
dtypes: float64(1), object(9)
memory usage: 2.2+ MB


In [177]:
df.shape

(26041, 10)

In [178]:
df.head(2)

1소속부서 결제방법           구분            대상인원      부서명 연번      집행금액  \
0  서울특별시장   카드  시정 관련 간담회 등        시장 등 12명  행정국 총무과  1  344000.0   
1  서울특별시장   카드  시정 관련 간담회 등  시장, 청년단체 등 38명  행정국 총무과  2  450000.0   

                집행목적              집행일시                     집행장소  
0     시정 정책추진 자문 간담회  2018-01-19 20:41    동신참치양재직(서초구 강남대로 213)  
1  시정 청년 정책 의견수렴 간담회  2018-01-23 19:06  서울시청구내식당 ( 중구 세종대로 110)

In [179]:
df.tail(2)

1소속부서 결제방법   구분        대상인원          부서명  연번      집행금액  \
12  안전총괄본부   카드  NaN  상황대응과 직원일동  안전총괄본부상황대응과  13  108000.0   
13  안전총괄본부   카드  NaN  상황대응과 직원일동  안전총괄본부상황대응과  14   52000.0   

                         집행목적              집행일시                집행장소  
12  송년 다과회를 위한 소요물품 구입비 지급(1)  2017-12-29 14:05      ㈜만점족(중구 태평로2가)  
13  송년 다과회를 위한 소요물품 구입비 지급(2)  2017-12-29 14:31  서울시청매점(중구 덕수궁길 15)

## Tidy 1차 : column 분리하기

In [180]:
# 집행장소와 주소를 분리하자
df_copy = df.copy()
df_copy['집행지주소'] = df_copy['집행장소'].str.extract(r'\((.*)\)')
df_copy['집행지주소'] = df_copy['집행지주소'].replace(' ', '')
df_copy['집행지명'] = df_copy['집행장소'].str.extract(r'(.*)\(')
df_copy['집행지명'] = df_copy['집행지명'].replace(' ', '')
del df_copy['집행장소']
df_copy.head(3)

1소속부서 결제방법           구분            대상인원      부서명 연번      집행금액  \
0  서울특별시장   카드  시정 관련 간담회 등        시장 등 12명  행정국 총무과  1  344000.0   
1  서울특별시장   카드  시정 관련 간담회 등  시장, 청년단체 등 38명  행정국 총무과  2  450000.0   
2  서울특별시장   카드  시정 관련 간담회 등         시장 등 8명  행정국 총무과  3  226000.0   

                       집행목적              집행일시         집행지주소       집행지명  
0            시정 정책추진 자문 간담회  2018-01-19 20:41  서초구 강남대로 213    동신참치양재직  
1         시정 청년 정책 의견수렴 간담회  2018-01-23 19:06   중구 세종대로 110  서울시청구내식당   
2  시정 주요 경제정책 현안추진 의견수렴 간담회  2018-01-29 20:13       종로구 북촌로        한뫼촌

In [181]:
# 필요없는 Column 없애기
del df_copy['연번']
del df_copy['부서명']
df_copy.head(3)

1소속부서 결제방법           구분            대상인원      집행금액  \
0  서울특별시장   카드  시정 관련 간담회 등        시장 등 12명  344000.0   
1  서울특별시장   카드  시정 관련 간담회 등  시장, 청년단체 등 38명  450000.0   
2  서울특별시장   카드  시정 관련 간담회 등         시장 등 8명  226000.0   

                       집행목적              집행일시         집행지주소       집행지명  
0            시정 정책추진 자문 간담회  2018-01-19 20:41  서초구 강남대로 213    동신참치양재직  
1         시정 청년 정책 의견수렴 간담회  2018-01-23 19:06   중구 세종대로 110  서울시청구내식당   
2  시정 주요 경제정책 현안추진 의견수렴 간담회  2018-01-29 20:13       종로구 북촌로        한뫼촌

In [182]:
# 집행일시를 년/월/일/시간으로 분리하자
df_copy['year'] = df_copy['집행일시'].str.extract(r'(20..)')
df_copy['month'] = df_copy['집행일시'].str.extract(r'-(\d\d)-')
df_copy['day'] = df_copy['집행일시'].str.extract(r'-(\d\d\s)')
df_copy['time'] = df_copy['집행일시'].str.extract(r'(\d?\d:\d?\d)')
df_copy.head(3)

1소속부서 결제방법           구분            대상인원      집행금액  \
0  서울특별시장   카드  시정 관련 간담회 등        시장 등 12명  344000.0   
1  서울특별시장   카드  시정 관련 간담회 등  시장, 청년단체 등 38명  450000.0   
2  서울특별시장   카드  시정 관련 간담회 등         시장 등 8명  226000.0   

                       집행목적              집행일시         집행지주소       집행지명  year  \
0            시정 정책추진 자문 간담회  2018-01-19 20:41  서초구 강남대로 213    동신참치양재직  2018   
1         시정 청년 정책 의견수렴 간담회  2018-01-23 19:06   중구 세종대로 110  서울시청구내식당   2018   
2  시정 주요 경제정책 현안추진 의견수렴 간담회  2018-01-29 20:13       종로구 북촌로        한뫼촌  2018   

  month  day   time  
0    01  19   20:41  
1    01  23   19:06  
2    01  29   20:13

In [183]:
del df_copy['집행일시']
df_copy.head(3)

1소속부서 결제방법           구분            대상인원      집행금액  \
0  서울특별시장   카드  시정 관련 간담회 등        시장 등 12명  344000.0   
1  서울특별시장   카드  시정 관련 간담회 등  시장, 청년단체 등 38명  450000.0   
2  서울특별시장   카드  시정 관련 간담회 등         시장 등 8명  226000.0   

                       집행목적         집행지주소       집행지명  year month  day   time  
0            시정 정책추진 자문 간담회  서초구 강남대로 213    동신참치양재직  2018    01  19   20:41  
1         시정 청년 정책 의견수렴 간담회   중구 세종대로 110  서울시청구내식당   2018    01  23   19:06  
2  시정 주요 경제정책 현안추진 의견수렴 간담회       종로구 북촌로        한뫼촌  2018    01  29   20:13

In [184]:
# '대변인' 테이블만 보기
df_copy[df_copy['1소속부서'] == '대변인'].head(3)

1소속부서 결제방법   구분       대상인원      집행금액                  집행목적     집행지주소  \
0   대변인   카드  NaN  신문팀장 등 5명   45000.0  언론모니터링 당직자 격려 간담회(1)  중구 을지로1길   
1   대변인   카드  NaN  신문팀장 등 5명   23000.0  언론모니터링 당직자 격려 간담회(2)    중구 무교로   
2   대변인   카드  NaN  대변인 등 11명  187000.0   언론 주요일정 수립 업무협의 간담회  중구 무교로16   

      집행지명  year month  day   time  
0  신가원설렁탕   2018    01  01   09:53  
1  커피빈코리아   2018    01  01   09:59  
2     참숯골   2018    01  02   12:34

## 파일로 저장하기

In [231]:
# 크롤링하는데 오래 걸리니 데이터를 파일로 저장해두자
from IPython.display import FileLink, FileLinks

df_copy.to_csv('C:\workspace\djangoPrj\popular_restaurants_from_officials/raw.csv', index=False)
# df_copy.to_excel('C:\workspace\djangoPrj\popular_restaurants_from_officials/seoul_dining_crawl.xlsx', index=False)

FileLinks('C:\workspace\djangoPrj\popular_restaurants_from_officials')

C:\workspace\djangoPrj\popular_restaurants_from_officials/
  official_div.json
  raw.csv
  README.md
  tidy_1.csv
  tidy_2.csv
  tidy_data.ipynb
  web_crawling.ipynb
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git/
  COMMIT_EDITMSG
  config
  description
  FETCH_HEAD
  HEAD
  index
  ORIG_HEAD
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\hooks/
  applypatch-msg.sample
  commit-msg.sample
  fsmonitor-watchman.sample
  post-update.sample
  pre-applypatch.sample
  pre-commit.sample
  pre-push.sample
  pre-rebase.sample
  pre-receive.sample
  prepare-commit-msg.sample
  update.sample
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\info/
  exclude
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\logs/
  HEAD
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\logs\refs\heads/
  master
  top10_release
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\logs\refs\remotes\origin/
  master
  top10_release
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\01/
  a688294847d7286d9359d325a03443552661b4
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\02/
  b0bbde2ec388bf85317b5d8a87db71f8a75145
  f258f9289a5382860b61dc0970673704cd36c6
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\03/
  357ce4f5833006114f2e790a7b89dbf69d4f90
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\04/
  73643c419af10e77427f5841c8d773d513e538
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\05/
  f3a6e5eb189936b17e93f28e86dc781e800f04
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\06/
  cfe42c68a7e5255cd1171e9ec4e1897a8678b9
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\09/
  7a86c699a42ed9e5b071b3b98cdee0abb9eafc
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\0a/
  096502da9b95076187df64b7f06376429c4a17
  a4b144b7d9d0cc947e4efe298b1fade7eb3c14
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\0b/
  cad6b506bd1e18b0144883d7f383e469077ef0
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\0c/
  bda561b9598626ea42cd5453b3ef69e69f3c0c
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\0d/
  070d2af1cb403b96adec78f49f9e14d05914c6
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\0f/
  435abcc307c02f8583856c9d37400a6d9965f5
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\10/
  a9e05be7c8f4b9b63fa1b8a4492cf9dd6e67ea
  f525538bfd568f87e3f54e156bed18f94c599a
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\13/
  c56b2a73f323af6eab968f93e05500b0c1a875
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\15/
  60c7b4a97586facb0a365183d872a2cc2db400
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\17/
  690a3478dae152fb4599c17ec97adbef5d7d69
  d1ad67cdcca17f6ddcdbb4edf062a9f2b49b60
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\18/
  4df53b1b936a65f7fb225ac51761227471a311
  4df9240ce76e1508086ac5bef6600c81809643
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\19/
  d1ab0ffef2b833068263d11143ccfa844d3462
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\1a/
  14959bcadac9d036d4d3296177437a94a884f4
  d2537d0b45e87ca9216d6377b6a78f28b04b20
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\1b/
  5880df5d6bb42432161b35b26f151c7a7a9320
  771fb608f2eea1516ad2ea3674d5d0f715ea17
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\1c/
  6b99fc0946c3f41df99174e3621eb88d3c23e7
  a64ae5b08ed18efda27c9a58a8496d31afac2a
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\1d/
  e874cf891a8fcf97097b49fb78762a5332eb87
C:\workspace\djangoPrj\popular_restaurants_from_officials\.git\objects\1e/
  6a57c825269bafb203f78db1a8754de8a07838
C:\workspa

## Top20 집계

In [232]:
df_copy = pd.read_csv('raw.csv')

In [233]:
df_copy.shape

(21260, 13)

In [241]:
df_copy['cnt'] = 1
df_copy.groupby('집행지명').sum().sort_values(by='cnt', ascending=False)[:5]

집행금액      year   month     day  cnt
집행지명                                           
참숯골   47375280.0  758606.0  2292.0  6126.0  376
우도일식  51429800.0  758610.0  2346.0  5898.0  376
구이구이  30290300.0  651665.0  1996.0  4981.0  323
곰국시집  34438350.0  593149.0  1955.0  4769.0  294
동해일식  38199000.0  472150.0  1300.0  3730.0  234

- 참숯골, 우도일식, 구이구이, 곰국시집이 여러개다. 왜지...?

In [216]:
name_arr = df_copy['집행지명'].unique()
np.set_printoptions(threshold=np.nan)
name_arr

array(['동신참치양재직', '서울시청구내식당 ', '한뫼촌', '서울시청구내식당', '삼다도', '（주）신화푸드긴자',
       '어상', '마쯔', '（주）달개비자연음식', '주식회사거구상사', '자연채샤브', '태진복집', '어부가',
       '일품당샤브샤브', '철판＆부대찌개브라더', '메이탄', '주식회사\u3000동원브이아', '화동식당',
       '주식회사서여의도해담', '동해수산', '림왕돈까스', '（주）신한디엠\u3000화로', '（주）식객촌', '훈민정',
       '（주）바닷가작은부엌', '남원추어탕', '노다지', '몽중헌\u3000안국점', '제주본가', '（주）베이징코야',
       '재단법인\u3000세종문화회', '통인가게', '피자헛독립문점', '（주）미스터피자\u3000청', '삼우정',
       '（주）웰리브', '가진화랑（갤러리카페', '씨유바비엥점', '한국피자헛\u3000약수２호', nan,
       '더\u3000후라이팬 ', '주식회사\u3000콩두에프앤 ', '한국피자헛 ', '스트릿\u3000연세세브란스 ',
       '수 ', '우정사업본부 ', '미스터피자목동점 ', '남도한식(정든님） ', '공미 ', '（주）터하집 ',
       '산낙지갯마을 ', '피자헛\u3000성내２호점 ', '미스터피자양재 ', '경복궁관훈점 ', '피자헛독립문점 ',
       '미스터피자수유점 ', '철판＆부대찌개브라더 ', '㈜디에스크린', '호텔그레이스뷰 ', '（주）씨앤피프랜즈 ',
       '처마끝하늘풍경 ', '돼지마을 ', '본도시락\u3000단국대점 ', '지혜 ', '오양식관 ',
       '（주）달개비자연음식 ', '（주）웰리브 ', '대보름 ', '빚은시청점 ', '광화문\u3000한옥집 ', '빌즈 ',
       '동해수산 ', '（주）미스터피자 ', '남포면옥 ', '（주）신한디엠\u3000화로 ',
       '주식회사\u3000동원브이아 ', '동해일

unique로 찍어보면, 아직 띄어쓰기가 제대로 제거되지 않았다.
<br>
```df_copy['집행지명'] = df_copy['집행지명'].replace(' ', '')```
를 수행해주었는데도 그렇네... 
<br>
`python replace space not working`으로 검색해보았더니 [이런](https://stackoverflow.com/questions/4369159/replace-n-with-br) 답변을 찾아볼 수 있었다. 

str을 해주라는 말인듯?

In [217]:
df_copy['집행지명'] = df_copy['집행지명'].str.replace(' ', '')

In [218]:
name_arr = df_copy['집행지명'].unique()
name_arr

array(['동신참치양재직', '서울시청구내식당', '한뫼촌', '삼다도', '（주）신화푸드긴자', '어상', '마쯔',
       '（주）달개비자연음식', '주식회사거구상사', '자연채샤브', '태진복집', '어부가', '일품당샤브샤브',
       '철판＆부대찌개브라더', '메이탄', '주식회사\u3000동원브이아', '화동식당', '주식회사서여의도해담',
       '동해수산', '림왕돈까스', '（주）신한디엠\u3000화로', '（주）식객촌', '훈민정', '（주）바닷가작은부엌',
       '남원추어탕', '노다지', '몽중헌\u3000안국점', '제주본가', '（주）베이징코야',
       '재단법인\u3000세종문화회', '통인가게', '피자헛독립문점', '（주）미스터피자\u3000청', '삼우정',
       '（주）웰리브', '가진화랑（갤러리카페', '씨유바비엥점', '한국피자헛\u3000약수２호', nan,
       '더\u3000후라이팬', '주식회사\u3000콩두에프앤', '한국피자헛', '스트릿\u3000연세세브란스', '수',
       '우정사업본부', '미스터피자목동점', '남도한식(정든님）', '공미', '（주）터하집', '산낙지갯마을',
       '피자헛\u3000성내２호점', '미스터피자양재', '경복궁관훈점', '미스터피자수유점', '㈜디에스크린',
       '호텔그레이스뷰', '（주）씨앤피프랜즈', '처마끝하늘풍경', '돼지마을', '본도시락\u3000단국대점', '지혜',
       '오양식관', '대보름', '빚은시청점', '광화문\u3000한옥집', '빌즈', '（주）미스터피자', '남포면옥',
       '동해일식', '모어\u3000댄\u3000레스（ＭＯ', '일일향（마포점）', '바루', '방이삿포로',
       '망원시장상인회', '무교소호정', '마마스', '혼신마켓\u3000광화문점', '롯데쇼핑주식회사', '구이구이',
       '물자리', '수리골쉼터', '마농초콜라띠

제대로 띄어쓰기가 없어졌다!
다시 top20을 내보자

In [240]:
df_copy.groupby('집행지명').sum().sort_values(by='cnt', ascending=False)[:5]

집행금액      year   month     day  cnt
집행지명                                           
참숯골   47375280.0  758606.0  2292.0  6126.0  376
우도일식  51429800.0  758610.0  2346.0  5898.0  376
구이구이  30290300.0  651665.0  1996.0  4981.0  323
곰국시집  34438350.0  593149.0  1955.0  4769.0  294
동해일식  38199000.0  472150.0  1300.0  3730.0  234

- '파리바게[트뜨]', '스타벅스'는 빼야겠다
- '시청'이 들어가는 곳도 빼야겠어
- year, month, day는 빼자

In [220]:
df_copy = df_copy[df_copy['집행지명'].str.contains('파리바게[트뜨]') == False]
df_copy = df_copy[df_copy['집행지명'].str.contains('시청') == False]
df_copy = df_copy[df_copy['집행지명'].str.contains('스타벅스') == False]

In [239]:
df_copy.groupby('집행지명')['집행지명', '집행금액', 'cnt'].sum().sort_values(by='cnt', ascending=False)[:5]

집행금액  cnt
집행지명                 
참숯골   47375280.0  376
우도일식  51429800.0  376
구이구이  30290300.0  323
곰국시집  34438350.0  294
동해일식  38199000.0  234

<br>
30위까지 찍어보니 '카페'/'커피' 키워드도 없어도 될 것 같다. 

In [222]:
df_copy = df_copy[df_copy['집행지명'].str.contains('카페') == False]
df_copy = df_copy[df_copy['집행지명'].str.contains('커피') == False]

In [250]:
df_copy.groupby('집행지명')['집행지명', '집행금액', 'cnt'].sum().sort_values(by='cnt', ascending=False)[:10]

집행금액  cnt
집행지명                    
참숯골      47375280.0  376
우도일식     51429800.0  376
구이구이     30290300.0  323
곰국시집     34438350.0  294
동해일식     38199000.0  234
무교소호정    19138000.0  181
바닷가작은부엌  26757000.0  173
동원참치     20246000.0  161
오리마당     17900550.0  160
창고43     20972300.0  157

In [251]:
df_copy.groupby('집행지명')['집행지명', '집행금액', 'cnt'].sum().sort_values(by='집행금액', ascending=False)[:10]

집행금액  cnt
집행지명                    
우도일식     51429800.0  376
참숯골      47375280.0  376
동해일식     38199000.0  234
곰국시집     34438350.0  294
구이구이     30290300.0  323
삼우정      28880000.0  143
만복림      27286000.0  149
바닷가작은부엌  26757000.0  173
더미       25462000.0   77
우정사업본부   24860000.0    6

In [225]:
df_copy.to_csv('tidy_1.csv', index=False)

이상하다. 
서울시장 직속 부서만 크롤링 했을 때
참숯골이 cnt가 400가 넘었는데
전 부서 데이터로 분석했는데 
376회로 줄어들었다.

지난번 데이터로 해보자

In [235]:
df_data_before = pd.read_csv('C:/Users/jeeyun/Downloads/crawl_data.csv')

In [249]:
df_data_before.groupby('집행지명')['집행지명', '집행금액', 'cnt'].sum().sort_values(by='cnt', ascending=False)[:10]

집행금액  cnt
집행지명                  
참숯골      42374960  406
우도일식     37630000  306
구이구이     22905800  246
곰국시집     17041300  178
바닷가작은부엌  24440400  168
동해일식     20206000  158
한가람      16370000  156
동해수산     17876000  144
무교소호정    14628000  138
제일가든     15924000  136

In [242]:
print('2017.8-2018.7전부서')
print(df_copy.shape)
print('2017년 직속부서')
print(df_data_before.shape)

전부서
(21260, 13)
직속부서
(15192, 12)


기간이 달라졌지용! 기록기록